docs/scenarios.md

```diff
@@ -0,0 +1,601 @@
+# Scenarios
+
+Scenarios for Peacock.
+
+## Local / Remote Settings Scenarios
+
+This set of scenarios clarifies Peacock's intended behavior when in a remote or local environment, and how it affects:
+
+- user and workspace settings for "peacock.color" and "peacock.remoteColor"
+- workspace colorCustomizations
+
+### A - Local / No User Settings / No Workspace Settings - [User Sets Local Color]
+
+```json
+// user settings
+{}
+```
+
+```json
+// workspace settings
+{}
+```
+
+- open a project
+- Peacock sees we are in local env., so it looks for workspace, then user settings for a "peacock.color". None exist, so we do nothing.
+- User changes the color to #00ff00 (green)
+- Peacock applies color customizations using #00ff00 (green)
+- User changed the color so we want to write it.
+- Peacock checks if "peacock.color" in workspace has changed. It has. So "peacock.color" is written to workspace.
+
+```json
+// user settings
+{}
+```
+
+```json
+// workspace settings
+{
+  "peacock.color": "#00ff00"
+}
+```
+
+### B - Local / No User Settings / Workspace Color - [User Sets Local Color]
+
+```json
+// user settings
+{}
+```
+
+```json
+// workspace settings
+{
+  "peacock.color": "#00ff00"
+}
+```
+
+- open a project
+- Peacock sees we are in local env., so it looks for workspace, then user settings for a "peacock.color".
+- Local workspace has "peacock.color" #00ff00
+- Peacock applies color customizations using #00ff00 (green)
+- Peacock checks if "peacock.color" in workspace is green. It is. So "peacock.color" is not written.
+
+```json
+// user settings
+{}
+```
+
+```json
+// workspace settings
+{
+  "peacock.color": "#00ff00"
+}
+```
+
+- User changes the color to "#ffff00" (yellow)
+- Peacock applies color customizations using #ffff00 (yellow)
+- User changed the color so we want to write it.
+- Peacock checks if "peacock.color" in workspace is yellow. It is not. So "peacock.color" is written to workspace.
+
+```json
+// user settings
+{
+  "peacock.color": "#00ff00"
```

Comment:
I would have expected that at this point the user settings are still empty. 

The invariant is that when the user customizes a peacock color using the peacock pickers, then this does not affect the user settings but only the workspace settings. The user has customized the workspace color, not the default peacock color.

The user settings can only be changed by

Reply:
yes, this was a typo.

fixed.